src/engine/threaded_engine.cc

```diff
@@ -403,7 +417,11 @@ inline void ThreadedEngine::OnComplete(ThreadedOpr* threaded_opr) {
   }
   // Mark complete for write variables.
   for (auto&& i : threaded_opr->mutable_vars) {
-    const bool debug_info = (engine_info_ && debug_wait_var_ == i);
+    if (threaded_opr->ex_ptr) {
+      i->ex_ptr = threaded_opr->ex_ptr;
+      if (!global_ex_ptr) global_ex_ptr = i->ex_ptr;
+    }
```

Comment:
suppose and operator has two outputs x and y and it raises an exception.
then x.asnumpy() would raise an error.
Then y.asnumpy() would raise the same error again.

Reply:
z += 1 won't execute since z already has an exception associated with it. z.asnumpy() will still raise the error.